packages/ipfs-core/src/components/name/publish.js

```diff
@@ -78,12 +78,12 @@ module.exports = ({ ipns, dag, peerId, isOnline, keychain }) => {
       throw err
     }
 
-    let pubLifetime
+    let pubLifetime = 0
     try {
-      pubLifetime = parseDuration(lifetime)
+      pubLifetime = parseDuration(lifetime) || 0
 
       // Calculate lifetime with nanoseconds precision
-      pubLifetime = pubLifetime.toFixed(6)
+      pubLifetime = Math.trunc(pubLifetime)
```

Comment:
These don't do the same thing?

```javascript
const parseDuration = require('parse-duration').default

const duration = '1.333333333h'
const parsed = parseDuration(duration)

console.info(parsed.toFixed(6))
// "4799999.998800"
console.info(Math.trunc(parsed))
// 4799999
```

Reply:
Thanks for catching this, I must have overlooked that `6` was passed there. It's just `toFixed()` returns string and not a number that is why I have changed this.

How about ?

```suggestion
      const precision = 1000000
      pubLifetime = Math.trunc(pubLifetime * precision) / precision
```
```